### PR TITLE
Feat/cypress render.tests

### PR DESCRIPTION
**Pull request by bentoper**
## What's the purpose of this pull request?
Adds e2e tests to make sure our pages are rendering. Tests Home, PDP and PLP.

## How to test it?
Using either a production or a dev build just run `yarn cy:open` and run the desired files or `yarn cy:run` to run all tests on the CLI. In either case the `baseUrl` on `cypress.json` should be the path to where the store is running.

### CODE REVIEW

File: cypress/e2e/render/testHome.tests.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+describe('check home page rendering', () => {
+  it('checks home', () => {
+    cy.visit('/')
+    cy.get('header')
+    cy.scrollTo('bottom')
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(1000)
```
Comment (Gmantiqueira) [inline cypress/e2e/render/testHome.tests.js:7]: Do we really need this cy.wait?

File: cypress/e2e/render/testPDP.tests.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+describe('check that PDP page is rendering', () => {
+  before(() => {
+    cy.visit('/')
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(1000)
```
Comment (Gmantiqueira) [inline cypress/e2e/render/testPDP.tests.js:5]: We don't need this one after the `cy.visit`.

File: cypress/e2e/render/testPDP.tests.js [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+    cy.get('[data-testid="productSummaryTitle"').eq(0).click()
```
Comment (Gmantiqueira) [inline cypress/e2e/render/testPDP.tests.js:6]: getById instead of cy.get

File: cypress/e2e/render/testPDP.tests.js [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  })
+  it('checks PDP renders and has a product to buy', () => {
+    cy.get('header').should('be.visible')
+    cy.get('[data-testid="addToCart"')
```
Comment (Gmantiqueira) [inline cypress/e2e/render/testPDP.tests.js:10]: getById instead of cy.get

File: cypress/e2e/render/testPLP.tests.js [hunk 5]
```diff
@@ -0,0 +1,15 @@
+describe('check that PLP page is rendering', () => {
+  before(() => {
+    cy.visit('/')
+    // eslint-disable-next-line cypress/no-unnecessary-waiting
+    cy.wait(1000)
```
Comment (Gmantiqueira) [inline cypress/e2e/render/testPLP.tests.js:5]: We don't need this one after the cy.visit.

File: cypress/e2e/render/testPLP.tests.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  })
+  it('checks PLP renders', () => {
+    cy.get('[data-testid="categoryLink"').eq(0).click()
```
Comment (Gmantiqueira) [inline cypress/e2e/render/testPLP.tests.js:8]: getById instead of cy.get.

File: cypress/e2e/render/testPLP.tests.js [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    cy.get('header').should('be.visible')
+    cy.get('[data-testid="productSummaryContainer"').end()
```
Comment (Gmantiqueira) [inline cypress/e2e/render/testPLP.tests.js:10]: getById instead of cy.get.

File: cypress/e2e/home/testLinks.test.js [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+      cy.intercept('GET', `/page-data${url.pathname}/page-data.json`).as(
+        `pageLoad${url.pathname}`
+      )
+      cy.visit($page.prop('href'), { failOnStatusCode: false })
+      cy.wait(`@pageLoad${url.pathname}`)
+        .its('response.statusCode')
+        .then(($code) => {
+          if ($code < 200 || $code > 399) {
+            cy.log(`Unexpected response code ${$code} for path ${url.pathname}`)
+          }
```
Comment (igorbrasileiro) [inline cypress/e2e/home/testLinks.test.js:33]: should use expect to assert?

**Review by igorbrasileiro — APPROVED** [review]

File: cypress/e2e/render/testHome.tests.js (hunk 1 above)
Comment (bentoper) [reply cypress/e2e/render/testHome.tests.js]: Nope

**Pull request merged**